src/faultlog/faultlog_main.cpp

```diff
@@ -119,16 +119,39 @@ void createNagPel(sdbusplus::bus::bus& bus,
     }
 }
 
+/** @brief Create faultlog pel for host poweron
+ *
+ *  As the application waits for host to reach running state,
+ *  re-read the guard records as they might be changed during IPL
+ *
+ *  @param[in] bus - D-Bus to attach to
+ */
+void createHostPowerOnNagPel(sdbusplus::bus::bus& bus)
+{
+    // Don't get ephemeral records because those type records are
+    // not intended to expose to the end user, just created for
+    // internal purpose to use by the BMC and Hostboot.
+    openpower::guard::GuardRecords records = openpower::guard::getAll(true);
```

Comment:
I see these lines in multiple places. Can we create a new function called getUnresolvedRecords(GuardRecords& unresolvedRecords) in utils?

Reply:
i will modify to add comments in one place only

Reply:
moved to new internal function as there is duplicate code